plugins/libplugin.c

```diff
@@ -651,9 +651,11 @@ static bool rpc_read_response_one(struct plugin *plugin)
 	bool valid;
 	const jsmntok_t *toks, *jrtok;
 
-	/* FIXME: This could be done more efficiently by storing the
-	 * toks and doing an incremental parse, like lightning-cli
-	 * does. */
+	/* For our convenience, lightningd always ends JSON requests with
```

Comment:
~~This forces us to print `\n\n` forevermore.  Maybe acceptable tho.  I think the incremental-parse idea would prevent us from effectively making `\n\n` part of the contract with the plugin.~~

~~If we are going to officially depend on this detail, I suggest adding a note in `doc/PLUGINS.md` or other documentation that our command stream acts this way.~~

~~I suppose my concern is that independent projects might attempt to run plugins by emulating the `lightningd` interface.  For example, the vaunted "fuzz testing" #2662.~~

EDIT: Nevermind you change it in a later commit.  Makes me wonder if **this** particular change needs to exist, tho.

Reply:
Turns out we already rely on this and publish it: in particular, the Python plugin library relies on it.

That's because most languages do not offer an interface for continuous parsing, and so are grossly inefficient :(

Reply:
Oh, and reworking it all is messy and means I have to rerun all my benchmarks... again :)